### PR TITLE
Migrate coincbc with coin_or_cbc

### PR DESCRIPTION
**Pull request by hmaarrfk**
Thanks

Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by hmaarrfk** [comment]
cc: @conda-forge/coin-or-cbc thanks for merging the run export upstream.

Is this in fact your preferred spelling?

File: recipe/linter_hints/hints.toml [hunk 1]
```diff
@@ -54,6 +54,12 @@ libgcc-ng = """\
         - {{ compiler("c") }}    # depending on which...
         - {{ compiler("cxx") }}  # ... compilers you use
     ```"""
+coincbc = """\
+    The `coincbc` package has been renamed to `coin-or-cbc`. It is now \
+    preferred to depend on the `coin-or-cbc` package in th host section and allow
+    the run_export define the runtime dependency.
+"""
```
Comment (hmaarrfk) [inline recipe/linter_hints/hints.toml:61]: see in particular this comment.

**Comment by tkralphs** [comment]
I'm not really sure what is meant by "preferred spelling." The name of the package is `coin-or-cbc`, so that is the spelling that has to be used. It is the package name that we chose intentionally if that is what you're asking. I don't know where the `coin_or_cbc` came from in the title here. 

Per the comment, it's not exactly the case that `coincbc` was renamed to `coin-or-cbc`. It's more accurate to say that `coin-or-cbc` is the officially supported package and `coincbc` was an unofficial package that was pre-existing and was essentially abandoned after `coin-or-cbc` came into existence. We merged `coincbc` into `coin-or-cbc` to avoid breaking dependencies for other packages. I don't fully understand the mechanics of how that was done.

**Comment by hmaarrfk** [comment]
> I don't know where the coin_or_cbc came from in the title here.

Alright, here are some fund details of conda-forge + yaml:

in yaml, you can't have dashes in the key names, so `coin-or-cbc` becomes `coin_or_cbc` in `recipe/conda_build_config.yaml`


>  coincbc was an unofficial package

This is a detail i'm not too sure downstream packagers care about. conda-forge, in my mind, was built around "unofficial packaging" so I'm 100% happy that the coin-or foundation is helping conda-forge, but probably best to leave it as a historical detail for historian ^_^


I'll change "preferred spelling" to preferred package name.

Ultimately, people having `coincbc` and `coin-or-cbc` in their environments is a little confusing when you are auditing things. This mostly helps people 'lint' their recipes so they continue to be searchable with github's interface.

Thank you very much for your feedback, input, and your hlep in keeping the package up to date!

**Comment by tkralphs** [comment]
OK, thanks for the explanation and sorry if I was snarky ;). I'm constantly reminded of how little I understand about conda, even after working with it for years. It's a complicated beast. I mostly bowed more knowledgeable maintainers when deciding how to handle the existence of `coincbc`. So far, it doesn't seem to cause confusion, but I don't mind getting rid of it either.

**Comment by hmaarrfk** [comment]
A few new changes happened:

1. this whole "migrate together" just came in a few weeks ago. so it became possible to do this.
2. I've been getting frustrated with `importlib-metadata` and `importlib_metadata` so I'm trying to help out with other packages that have "two names" by providing hints.

Thanks for your responsiveness on the feedstock, it really hurts to be "pushing" things to feedstocks we don't maintain as core, and the lack of the correct run_exports did hurt people.

Honestly, I had both PRs made in parallel, i didn't know if you would respond to either ....

**Pull request merged**